### ISSUE TITLE
[Getting Familiar][Refactor] list out hardcodings in the preprocessors

### ISSUE THREAD

**Issue by Udayraj123** [Udayraj123/OMRChecker#85]
**Describe the bug**
There are some hardcodings in the codebase(some of which are tagged by a todo comment). We need to remove them and pick them up from configuration/command line. The first step would be to list out the hardcoded values in a comment below. 
Then we'll break it down into new issues to fix them in parts.

**To Reproduce**
One example is from `src/processors/CropPage.py` : the parameters passed to `cv2.Canny` can be taken from something like `self.canny_params`

<img width="357" alt="image" src="https://user-images.githubusercontent.com/16881051/194460831-3c8fad06-2451-445d-be70-3de23d8f7951.png">

**Acceptance Criteria**
A list of all hardcodings should be populated.

**Screenshots**
NA

**Desktop (please complete the following information):**
 - OS: any
 - Python version: 3.10.6
 - OpenCV version: 4.6.0

**Additional context**
NA

**Comment by tuxo** [comment]
can i be assigned to this one? ☺️

**Comment by Udayraj123** [comment]
@tuxo any updates?

**Comment by GeorgiaMayDay** [comment]
Since @tuxo hasn't updated his progress in a while I'm going to have a lookthrough for hardcodings

**Comment by GeorgiaMayDay** [comment]
Since @tuxo hasn't updated this issue in over a month, I thought I'd try it

**Comment by GeorgiaMayDay** [comment]
Line 336 in core.py
Score should be automatically evaluated but it currently hardcoded as a constant
![image](https://user-images.githubusercontent.com/115997517/203036538-3af7e548-a8c4-4ad6-a1d9-47bd905ed802.png)

 
After line 25 in CropOnMarkers
Hardcode: relativePath: “Omr_marker.jpg” are both set up in the template json samples – could this not be gotten from there?
![image](https://user-images.githubusercontent.com/115997517/203036673-1c02d959-a2e3-4c4a-b4c9-38c0f1e3d63c.png)

 


From line 81 in CropPages
Hardcode: This could be passed in from the template configuration
![image](https://user-images.githubusercontent.com/115997517/203036735-d394ae19-644f-4b5a-81c6-c1ce18ffb266.png)
 
Line 120 in imgutils.py
Hardcode: All the variables here are hard coded when it feels like these would be better as defaults that the user could change in config
![image](https://user-images.githubusercontent.com/115997517/203036760-7bc91fba-50ef-4899-ad03-c2472b2cba13.png)

 
Line 627 in imagutil.py
Hardcode: both x and y are hard coded when they should either be constants or configurable
 
![image](https://user-images.githubusercontent.com/115997517/203036786-7b513e72-bfea-4d37-8e92-8c9684767f55.png)


Line 658 in imagutil.py
Hardcode: similar to the above, x and y have been hardcode. 
 
![image](https://user-images.githubusercontent.com/115997517/203036811-131b01f4-5b0b-41cd-a765-9629f77b8e3a.png)


Line 914 and 915 in imagutil.py
Hardcode: these should be constants or configurable
 
![image](https://user-images.githubusercontent.com/115997517/203036835-a59741c0-2382-48cb-9939-c9f6db9d76c2.png)

 

Other:

This is line is repeated in FeatureBasedAlignment, CropPages  and imgutils– perhaps worth putting into a separate file and importing around to keep the code DRY?
![image](https://user-images.githubusercontent.com/115997517/203036949-795e4348-269c-4f92-9d00-3e6b9e9cf737.png)

 
From 125 in core.py
There is some unreachable logger code due to the return True.
Suggestion: Either the early return True is deleted or the logger code
![image](https://user-images.githubusercontent.com/115997517/203036973-eee36da7-57dd-4993-8f52-61e3f48252e5.png)

**Comment by GeorgiaMayDay** [comment]
I noticed some other minor issues apart from hardcoding I highlighted under other

**Comment by Udayraj123** [comment]
Hi @GeorgiaMayDay, sorry I missed above comment. Will check soon. I hope you've joined over discord as well(I tend to respond quicker there)

**Comment by pranavajith** [comment]
I'm up for this challange, @Udayraj123 ..
Although some resources as to how to set hardcodings in preprocessors would be amazing.

**Comment by Udayraj123** [comment]
@pranavajith please check discord for my reply.

**Comment by Udayraj123** [comment]
Code is updated with some of the hardcodings reduced. New contributors can help here by checking for more hardcodings in the fresh code.

**Comment by esing1201** [comment]
Hi @Udayraj123 I am happy to help. I will appreciate some resources if possible.

**Comment by rakeshcj** [comment]
HI @Udayraj123 is the issue still open ?

**Comment by Udayraj123** [comment]
Hi @rakeshcj, yes the issue is still open. Most of the hardcodings are in pre-processors that still need attention. Do let me know if you're picking it up!

**Comment by rakeshcj** [comment]
Hi Udayraj,

I will work on it and send the pr.

Thanks for letting me know..

**Comment by Udayraj123** [comment]
Hi @rakeshcj let me know if any blockers.

**Comment by Darpana-iitgn** [comment]
Hi @Udayraj123, I am a beginner to open source, can I try working on this issue? I would appreciate any help/resources if possible

**Comment by Udayraj123** [comment]
Hey @Darpana-iitgn, sure! 
This task is now relevant for the v2 code since it contains a lot of refactoring (yet has hardcodings at places)
Please go through the dev branch code and try to find the hardcoded items. Based on each item you share I can provide a way to make it generic to be consumable from on of the schemas (template_schema.py etc).

**Comment by ryan1234814** [comment]
@Udayraj123  .Hi Sir , are there any issues that I can  contribute  to solve in   this project

**Comment by Udayraj123** [comment]
@ryan1234814 definitely, please check issues tagged with `good first issue` on this repository including this issue itself. You may feel comfortable to start with submitting a sample omr image as part of #40

**Comment by SRIVASTAVASRIHARSHA** [comment]
Hi, I'm new here. I'd like to work on listing the hardcodings for this issue. Could you please assign it to me?
